mempool/src/lib.rs

```diff
@@ -157,9 +178,14 @@ impl<'env> Mempool<'env> {
                     break;
                 }
 
+                let old_sender_account = sender_account.clone();
                 sender_account = sender_account
                     .with_outgoing_transaction(tx, block_height)
                     .expect("Failed to apply existing transaction");
+                if !state.filter.accepts_sender_account(&transaction, &old_sender_account, &sender_account) {
```

Comment:
I think this is the wrong place to do your check.
In this loop, we are applying other transactions.
You might want to move your check towards line ~200 (see my comment there).

Reply:
Ok, I cannot add comments in lines that haven't changed.
It is around `// Now, check the new transaction.` when we actually apply the current transaction to the Mempool.